### PR TITLE
Fix merkle tree

### PR DESCRIPTION
**Pull request by xiangxingxing**

### CODE REVIEW

File: test/AElf.Types.Tests/BinaryMerkleTestTest.cs [hunk 1]
```diff
@@ -0,0 +1,301 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace AElf.Types.Tests
+{
+    public class BinaryMerkleTestTest
```
Comment (aerkol) [inline test/AElf.Types.Tests/BinaryMerkleTestTest.cs]: Would it be better if we tested the number of nodes (count) is correct after generating a Merkle Tree. Like, from 4 leafs, 7 nodes; from 5 leafs, 13 nodes?

File: protobuf/aelf/core.proto [hunk 2]
```diff
@@ -89,3 +87,18 @@ message SInt64Value
...
+}
+
+message MerklePathNode{
+    Hash hash = 1;
+    bool isLeftChildNode = 2;
+}
+
+message BinaryMerkleTree {
+    repeated Hash nodes = 1;
+    Hash root = 2;
```
Comment (aerkol) [inline protobuf/aelf/core.proto]: The root is already the last element in nodes, do we need to duplicate it in message? Why don't we delete it (root) from proto declaration add it in partial implementation (BinaryMerkleTree.cs) instead?

`public Hash Root => Nodes.Any() ? Nodes.Last() : Hash.Empty;`

**Comment by shiwk** [comment]
@xiangxingxing Please check whether it is possible to define `BinaryMerkleTree` in a plain class, not generated by `protobuf`.

File: protobuf/aelf/core.proto (hunk 2 above)
Comment (xiangxingxing) [reply protobuf/aelf/core.proto]: We should make sure all the merkle tree nodes have been computed completely , in which case the expression above is valid.

**Pull request merged**